cli/src/main/java/org/eclipse/hono/cli/Commander.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * ******************************************************************************
+ *  * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *  *
+ *  * See the NOTICE file(s) distributed with this work for additional
+ *  * information regarding copyright ownership.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  ******************************************************************************
+ *
+ */
+
+package org.eclipse.hono.cli;
+
+import javax.annotation.PostConstruct;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.core.buffer.Buffer;
+import org.eclipse.hono.client.CommandClient;
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.util.BufferResult;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.*;
+
+/**
+ * Command application that connects to the Hono, get commands from the user using System.in, send those commands to devices
+ * and logs command responses if any.
+ */
+@Component
+@Profile("command")
+public class Commander extends AbstractClient {
+
+    private final Scanner scanner = new Scanner(System.in);
+    @Value(value = "${command.timeoutInSeconds}")
+    protected int requestTimeoutInSecs;
+    private WorkerExecutor workerExecutor;
+
+    /**
+     * Starts this component.
+     *
+     */
+    @PostConstruct
+    void start() {
+        workerExecutor = vertx.createSharedWorkerExecutor("user-input-pool", 3, TimeUnit.HOURS.toNanos(1));
```

Comment:
out of curiosity, what's wrong with the default worker threads provided by vert.x?

Reply:
Default maximum execution time of default worker threads is 6000 ms. In this usecase of getting input from user using System.in, I wanted to increase it. Of course it can be changed using the vertxOptions. Then this changed value will be used by all the default working threads and named worker executors (that may be created). So I created a separate named worker executor and changed the maximum execution time of it, so that it remains within this scope.